spec/urbanopt/urbanopt_scenario_reports_spec.rb

```diff
@@ -94,4 +94,46 @@
     
   end
 
+  it 'can merge construction costs together' do
+
+    existing_costs = []
+    new_costs = []
+   
+    new_costs << URBANopt::Scenario::DefaultReports::ConstructionCost.new({:category => "Construction", :item_name => "wall", :unit_cost => 1, 
+    :cost_units => "CostPerEach", :item_quantity => 1, :total_cost => 1})
+    new_costs << URBANopt::Scenario::DefaultReports::ConstructionCost.new({:category => "Construction", :item_name => "roof", :unit_cost => 1, 
+    :cost_units => "CostPerEach", :item_quantity => 1, :total_cost => 1})
+     
+    existing_costs << URBANopt::Scenario::DefaultReports::ConstructionCost.new({:category => "Construction", :item_name => "wall", :unit_cost => 1, 
+    :cost_units => "CostPerEach", :item_quantity => 1, :total_cost => 1})
+    existing_costs << URBANopt::Scenario::DefaultReports::ConstructionCost.new({ :category => "HVACComponent", :item_name => "hvac", :unit_cost => 1,
+    :cost_units => "CostPerEach", :item_quantity => 1, :total_cost => 1})
+    
+    URBANopt::Scenario::DefaultReports::ConstructionCost.merge_construction_costs(existing_costs, new_costs) 
+    
+  end
+
+  it 'can merge reporting periods together' do
+
+    existing_periods = []
+    new_periods = []
+    
+    existing_periods << {id: 5, name: "Annual", multiplier: 1, start_date: {month: 1, day_of_month: 1, year: 2019} , 
+    end_date: {month: 12, day_of_month:31, year: 2019 }, total_site_energy: 1, total_source_energy: 1, 
+    end_uses: { electricity: { heating: 1, cooling:1 }, natural_gas:{fans: 1, pumps: 1} } }
+    existing_periods << {id: 6, name: "January", multiplier: 1, start_date: {month: 1, day_of_month: 1, year: 2019} , 
+    end_date: {month: 1, day_of_month:31, year: 2019 }, total_site_energy: 1, total_source_energy: 1, 
+    end_uses: { electricity: { heating: 1, cooling:1 }, natural_gas:{fans: 1, pumps: 1} } }
+
+    new_periods << {id: 5, name: "Annual", multiplier: 1, start_date: {month: 1, day_of_month: 1, year: 2019} , 
+    end_date: {month: 12, day_of_month:31, year: 2019 }, total_site_energy: 1, total_source_energy: 1, 
+    end_uses: { electricity: { heating: 1, cooling:1 }, natural_gas:{fans: 1, pumps: 1} } }
+    new_periods << {id: 7, name: "February", multiplier: 1, start_date: {month: 1, day_of_month: 1, year: 2019} , 
```

Comment:
I think this case should issue a warning or maybe an error.  If one simulation didn't report out for January then the results are bad.

Reply:
Do you mean that the reporting periods for the features can not be different ?

Reply:
That was my understanding, lets discuss at 9:30